### PR TITLE
feat: DE-2 : add organization schema

### PR DESCRIPTION
**Pull request by shashankbaranwal**

### CODE REVIEW

**Comment by Nikhil-Rawat** [comment]
1. Use lowercase on the commit messagea.

File: src/repositories/organization/IOrganizationDocument.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import IVersionalbleDocument from "../versionable/IDocument";
+
+export default interface IOrganizationModel extends IVersionalbleDocument {
+  repository: IRepository;
+  public_repository: Number;
+  private_repository: Number;
+  createdAt: Date;
+  members: IMember;
```
[the remark this answers is not in the document]
Comment (shashankbaranwal) [reply src/repositories/organization/IOrganizationDocument.ts:8]: Resolve Both Point:-
1. Add member in the array.
2. Add repository in the array.

File: src/repositories/organization/IOrganizationDocument.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+interface IOwner {
+  userName: string;
+  type: string;
+}
+
+interface ICollaborator {
+  username: string;
+  permission: {
```
[the remark this answers is not in the document]
Comment (shashankbaranwal) [reply src/repositories/organization/IOrganizationDocument.ts:28]: We will be created interface for permission.

File: src/repositories/organization/OrganizationSchema.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
+import VersionableSchema from "../versionable/Schema";
+import { Schema } from "mongoose";
+
+const Owner = new Schema({
+  userName: {
```
[the remark this answers is not in the document]
Comment (shashankbaranwal) [reply src/repositories/organization/OrganizationSchema.ts:5]: Resolve it.

File: src/repositories/versionable/IDocument.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import * as mongoose from 'mongoose';
+
+export default interface IVersionableDocument extends mongoose.Document {
+
```
[the remark this answers is not in the document]
Comment (shashankbaranwal) [reply src/repositories/versionable/IDocument.ts:4]: Resolve it.

File: src/repositories/organization/OrganizationSchema.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  }
+});
+
+const Permission = new Schema({
+  admin: Boolean,
+  push: Boolean,
+  pull: Boolean,
+})
+
+const Collaborator = new Schema({
```
[the remark this answers is not in the document]
Comment (shashankbaranwal) [reply src/repositories/organization/OrganizationSchema.ts:17]: Resolve it.

File: src/repositories/organization/OrganizationSchema.ts [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+import { Schema } from "mongoose";
+
+const Owner = new Schema({
+  userName: {
+    type: String,
+    required: true,
+  }
+});
+
+const Permission = new Schema({
```
[the remark this answers is not in the document]
Comment (shashankbaranwal) [reply src/repositories/organization/OrganizationSchema.ts:11]: Resolve it

File: src/repositories/organization/OrganizationSchema.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (shashankbaranwal) [reply src/repositories/organization/OrganizationSchema.ts:17]: Resolve it

File: src/repositories/organization/OrganizationSchema.ts [hunk 7]
```diff
@@ -0,0 +1,79 @@
+import VersionableSchema from "../versionable/Schema";
```
[the remark this answers is not in the document]
Comment (shashankbaranwal) [reply src/repositories/organization/OrganizationSchema.ts:1]: Resolve it

File: src/repositories/organization/OrganizationRepository.ts [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+  constructor() {
+    super(organizationModel);
+  }
+
+  public static generateObjectId() {
+    return String(mongoose.Types.ObjectId());
+  }
+
+  public async create(data: any): Promise<IOrganizationDocument> {
+    const id = UserRepository.generateObjectId();
```
Comment (barkhatyagi) [inline src/repositories/organization/OrganizationRepository.ts:18]: add interface of data

**Review by Nikhil-Rawat — CHANGES_REQUESTED** [review]
add comment

**Review by Nikhil-Rawat — APPROVED** [review]

**Review by Nikhil-Rawat — APPROVED** [review]

**Review by Nikhil-Rawat — APPROVED** [review]

**Review by Nikhil-Rawat — APPROVED** [review]

**Review by barkhatyagi — APPROVED** [review]

**Pull request merged**